sphinx/environment/__init__.py

```diff
@@ -184,11 +184,19 @@ def __init__(self, app: Sphinx) -> None:
         # docnames to re-read unconditionally on next build
         self.reread_always: set[str] = set()
 
-        # docname -> pickled doctree
         self._pickled_doctree_cache: dict[str, bytes] = {}
+        """In-memory cache for reading pickled doctrees from disk.
```

Comment:
Ah, can you just keep the 'docname -> pickled doctree' comment so that we know what the key is ?

Reply:
sure, although I don't think the previous comments were particularly useful 😏; had no idea what they were used for

Reply:
It's more for implementation details and debugging IMO.

Reply:
FYI, for implementation, I just made this graph to map out the sphinx build process 😉  https://sphinx-process-graph.readthedocs.io

Reply:
Oh, I usually prefer using the old bruteforce approach via `prints`. I more or less know the flow but not necessarily what the objects hold (this is the main reason why I like having the structure in the comment before printing things).